### PR TITLE
[keyboard] Add z70ultra_v1 to MelGeek!

### PR DESCRIPTION
**Pull request by mj-bomb**
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/melgeek/z70ultra_v1/readme.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
+# Z70 Ultra v1
+
+![z70ultra](https://drive.google.com/file/d/1m8Qra8L7Nq4I-ewq1_GkKi9j0LVPN7_l/view?usp=sharing)
```
Comment (dunk2k) [inline keyboards/melgeek/z70ultra_v1/readme.md:3]: 1. url is not image file
2. can image be re-hosted in Imgur

File: keyboards/melgeek/z70ultra_v1/keymaps/via/keymap.c [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+        _______, RGB_SPI, RGB_SPD, _______, _______, _______, _______, _______, _______, _______, _______, _______,          EE_CLR,  _______,
+        _______,          _______, _______, _______, _______, _______, NK_TOGG, _______, _______, _______, _______, _______, KC_VOLU, _______,
+        _______, _______, _______,          		 _______, 				             _______, _______,          KC_MPRV, KC_VOLD, KC_MNXT
+    ),
+	[2] = LAYOUT_65_ansi_blocker( /* MO(2) */
+        _______, KC_BRID, KC_BRIU, KC_MCTL, KC_LPAD, RGB_VAD, RGB_VAI, KC_MPRV, KC_MPLY, KC_MNXT, KC_MUTE, KC_VOLD, KC_VOLU, _______, _______,
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______, _______,
+        _______,          _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
+        _______, _______, _______,					 _______,						     _______, _______,          _______, _______, _______
```
Comment (dunk2k) [inline keyboards/melgeek/z70ultra_v1/keymaps/via/keymap.c:38]:
```suggestion
        KC_LCTL, KC_LGUI, KC_LALT,		     KC_SPC,                             MO(1),   MO(2),            KC_LEFT, KC_DOWN, KC_RGHT
    ),
    [1] = LAYOUT_65_ansi_blocker( /* MO(1) */
        KC_GRV,  KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,  KC_DEL,  _______,
        _______, RGB_TOG, RGB_MOD, RGB_HUI, RGB_HUD, RGB_SAI, RGB_SAD, RGB_VAI, RGB_VAD, _______, KC_PSCR, KC_SCRL, KC_PAUS, QK_BOOT, _______,
        _______, RGB_SPI, RGB_SPD, _______, _______, _______, _______, _______, _______, _______, _______, _______,          EE_CLR,  _______,
        _______,          _______, _______, _______, _______, _______, NK_TOGG, _______, _______, _______, _______, _______, KC_VOLU, _______,
        _______, _______, _______,          	     _______, 				 _______, _______,          KC_MPRV, KC_VOLD, KC_MNXT
    ),
	[2] = LAYOUT_65_ansi_blocker( /* MO(2) */
        _______, KC_BRID, KC_BRIU, KC_MCTL, KC_LPAD, RGB_VAD, RGB_VAI, KC_MPRV, KC_MPLY, KC_MNXT, KC_MUTE, KC_VOLD, KC_VOLU, _______, _______,
        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______, _______,
        _______,          _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
        _______, _______, _______,		     _______,			         _______, _______,          _______, _______, _______ 
```

File: keyboards/melgeek/z70ultra_v1/info.json [hunk 3]
```diff
@@ -0,0 +1,317 @@
...
+        "vid": "0xEDED",
+        "pid": "0x6575",
+        "device_version": "0.0.1"
+    },
+	"processor": "STM32F303",
+    "bootloader": "stm32-dfu",
+    "board": "QMK_PROTON_C",
+	"bootmagic": {
+        "matrix": [0, 0]
+    },
```
Comment (dunk2k) [inline keyboards/melgeek/z70ultra_v1/info.json:16]: [0 , 0] is default settings for bootmagic
```suggestion
```

File: keyboards/melgeek/z70ultra_v1/info.json [hunk 4]
```diff
@@ -0,0 +1,317 @@
...
+			{ "flags": 4, "matrix": [4, 5],  "x": 40,  "y": 40 },
+            { "flags": 4, "matrix": [4, 5],  "x": 50,  "y": 40 },
+			{ "flags": 4, "matrix": [4, 5],  "x": 60,  "y": 40 },          
+            { "flags": 4, "matrix": [4, 9],  "x": 90,  "y": 40 },
+            { "flags": 4, "matrix": [4, 10], "x": 100, "y": 40 },
+            { "flags": 4, "matrix": [4, 12], "x": 120, "y": 40 },
+            { "flags": 4, "matrix": [4, 13], "x": 130, "y": 40 },
+            { "flags": 4, "matrix": [4, 14], "x": 140, "y": 40 }
+        ]
+    },
```
Comment (dunk2k) [inline keyboards/melgeek/z70ultra_v1/info.json:162]: conforms to PR guidelines
```suggestion
    },
    "layout_aliases": {
        "LAYOUT_all": "LAYOUT_ansi_split_space"
    },
```

**Review by dunk2k — CHANGES_REQUESTED** [review]

**Comment by dunk2k** [comment]
missing a `rules.mk` file in `keyboards/melgeek/z70ultra_v1` directory with contents of:
```
DEFAULT_FOLDER = melgeek/z70ultra_v1/rev1

```

File: keyboards/melgeek/z70ultra_v1/info.json [hunk 5]
```diff
@@ -0,0 +1,317 @@
...
+    "community_layouts": ["65_ansi_blocker"],
+    "layouts": {
```
Comment (dunk2k) [inline keyboards/melgeek/z70ultra_v1/info.json:164]: As per PR guidelines, would you be able to add a `LAYOUT_all` entry in `layouts`

**Review by dunk2k — CHANGES_REQUESTED** [review]

**Comment by zvecr** [comment]
> You must stop closing/opening new PRs for this same keyboard. It fragments the review process, also making it take longer.
>
> Review suggestions and any other updates should just be pushed to the same branch.

And _yet another_ PR, this continued behaviour will likely get you banned from the repo.

**Comment by mj-bomb** [comment]
> > You must stop closing/opening new PRs for this same keyboard. It fragments the review process, also making it take longer.
> > Review suggestions and any other updates should just be pushed to the same branch.
> 
> And _yet another_ PR, this continued behaviour will likely get you banned from the repo.

I'm sorry, I haven't used it in a long time. I didn't understand you before. It won't happen again

File: keyboards/melgeek/z70ultra_v1/readme.md (hunk 1 above)
Comment (mj-bomb) [reply keyboards/melgeek/z70ultra_v1/readme.md:3]: yes

File: keyboards/melgeek/z70ultra_v1/info.json (hunk 5 above)
Comment (mj-bomb) [reply keyboards/melgeek/z70ultra_v1/info.json:164]: hello, I'm not sure if that's true?

File: keyboards/melgeek/z70ultra_v1/config.h [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+#define RGB_MATRIX_DEFAULT_MODE RGB_MATRIX_CYCLE_ALL
+#define DRIVER_ADDR_1 0b0110000
+#define DRIVER_COUNT 1
+#define	RGB_MATRIX_LED_COUNT 	75
+#define CAPS_LOCK_INDEX_LEFT 	71
+#define CAPS_LOCK_INDEX_RIGHT 	72
+#define NUM_LOCK_INDEX_LEFT 	70
+#define NUM_LOCK_INDEX_RIGHT 	73
+#define SCROLL_LOCK_INDEX_LEFT 	69
+#define SCROLL_LOCK_INDEX_RIGHT 74
```
Comment (drashna) [inline keyboards/melgeek/z70ultra_v1/config.h:33]: mix of tabs and spaces :(
```suggestion
#define	RGB_MATRIX_LED_COUNT 75
#define CAPS_LOCK_INDEX_LEFT 71
#define CAPS_LOCK_INDEX_RIGHT 72
#define NUM_LOCK_INDEX_LEFT 70
#define NUM_LOCK_INDEX_RIGHT 73
#define SCROLL_LOCK_INDEX_LEFT 69
#define SCROLL_LOCK_INDEX_RIGHT 74
```

File: keyboards/melgeek/z70ultra_v1/rev1/rev1.c [hunk 7]
```diff
@@ -0,0 +1,145 @@
...
+	
+	if (host_keyboard_led_state().scroll_lock) {
+        RGB_MATRIX_INDICATOR_SET_COLOR(SCROLL_LOCK_INDEX_LEFT, 0, 0, 0xff);
+		RGB_MATRIX_INDICATOR_SET_COLOR(SCROLL_LOCK_INDEX_RIGHT, 0, 0, 0xff);
+    } else {
+		RGB_MATRIX_INDICATOR_SET_COLOR(SCROLL_LOCK_INDEX_LEFT, 0, 0, 0);
+		RGB_MATRIX_INDICATOR_SET_COLOR(SCROLL_LOCK_INDEX_RIGHT, 0, 0, 0);
+    }
+	
+	return false;
```
Comment (drashna) [inline keyboards/melgeek/z70ultra_v1/rev1/rev1.c:132]:
```suggestion
        return false;
    }

    if (host_keyboard_led_state().caps_lock) {
        RGB_MATRIX_INDICATOR_SET_COLOR(CAPS_LOCK_INDEX_LEFT, 0xff, 0, 0);
        RGB_MATRIX_INDICATOR_SET_COLOR(CAPS_LOCK_INDEX_RIGHT, 0xff, 0, 0);
    } else {
        RGB_MATRIX_INDICATOR_SET_COLOR(CAPS_LOCK_INDEX_LEFT, 0, 0, 0);
        RGB_MATRIX_INDICATOR_SET_COLOR(CAPS_LOCK_INDEX_RIGHT, 0, 0, 0);
    }
	
    if (host_keyboard_led_state().num_lock) {
        RGB_MATRIX_INDICATOR_SET_COLOR(NUM_LOCK_INDEX_LEFT, 0, 0xff, 0);
        RGB_MATRIX_INDICATOR_SET_COLOR(NUM_LOCK_INDEX_RIGHT, 0, 0xff, 0);
    } else {
        RGB_MATRIX_INDICATOR_SET_COLOR(NUM_LOCK_INDEX_LEFT, 0, 0, 0);
        RGB_MATRIX_INDICATOR_SET_COLOR(NUM_LOCK_INDEX_RIGHT, 0, 0, 0);
    }
	
    if (host_keyboard_led_state().scroll_lock) {
        RGB_MATRIX_INDICATOR_SET_COLOR(SCROLL_LOCK_INDEX_LEFT, 0, 0, 0xff);
        RGB_MATRIX_INDICATOR_SET_COLOR(SCROLL_LOCK_INDEX_RIGHT, 0, 0, 0xff);
    } else {
        RGB_MATRIX_INDICATOR_SET_COLOR(SCROLL_LOCK_INDEX_LEFT, 0, 0, 0);
        RGB_MATRIX_INDICATOR_SET_COLOR(SCROLL_LOCK_INDEX_RIGHT, 0, 0, 0);
    }
	
    return false;
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/melgeek/z70ultra_v1/readme.md (hunk 1 above)
Comment (dunk2k) [inline keyboards/melgeek/z70ultra_v1/readme.md:3]:
```suggestion
![z70ultra](https://i.imgur.com/MzATZCOh.jpg)
```

**Review by dunk2k — CHANGES_REQUESTED** [review]

File: keyboards/melgeek/z70ultra_v1/info.json [hunk 8]
```diff
@@ -0,0 +1,314 @@
...
+			{ "flags": 4, "matrix": [4, 6],  "x": 60,  "y": 40 },          
+            { "flags": 4, "matrix": [4, 9],  "x": 90,  "y": 40 },
+            { "flags": 4, "matrix": [4, 10], "x": 100, "y": 40 },
+            { "flags": 4, "matrix": [4, 12], "x": 120, "y": 40 },
+            { "flags": 4, "matrix": [4, 13], "x": 130, "y": 40 },
+            { "flags": 4, "matrix": [4, 14], "x": 140, "y": 40 }
+        ]
+    },
+    "community_layouts": ["65_ansi_blocker"],
+    "layouts": {
```
Comment (dunk2k) [inline keyboards/melgeek/z70ultra_v1/info.json:161]: `LAYOUT_all` added
```suggestion
    "layouts": {
        "LAYOUT_all": {
            "layout": [
                {"matrix": [0, 0], "x":0, "y":0},
                {"matrix": [0, 1], "x":1, "y":0},
                {"matrix": [0, 2], "x":2, "y":0},
                {"matrix": [0, 3], "x":3, "y":0},
                {"matrix": [0, 4], "x":4, "y":0},
                {"matrix": [0, 5], "x":5, "y":0},
                {"matrix": [0, 6], "x":6, "y":0},
                {"matrix": [0, 7], "x":7, "y":0},
                {"matrix": [0, 8], "x":8, "y":0},
                {"matrix": [0, 9], "x":9, "y":0},
                {"matrix": [0, 10], "x":10, "y":0},
                {"matrix": [0, 11], "x":11, "y":0},
                {"matrix": [0, 12], "x":12, "y":0},
                {"matrix": [0, 13], "x":13, "y":0, "w":2},
                {"matrix": [0, 14], "x":15, "y":0},

                {"matrix": [1, 0], "x":0, "y":1, "w":1.5},
                {"matrix": [1, 1], "x":1.5, "y":1},
                {"matrix": [1, 2], "x":2.5, "y":1},
                {"matrix": [1, 3], "x":3.5, "y":1},
                {"matrix": [1, 4], "x":4.5, "y":1},
                {"matrix": [1, 5], "x":5.5, "y":1},
                {"matrix": [1, 6], "x":6.5, "y":1},
                {"matrix": [1, 7], "x":7.5, "y":1},
                {"matrix": [1, 8], "x":8.5, "y":1},
                {"matrix": [1, 9], "x":9.5, "y":1},
                {"matrix": [1, 10], "x":10.5, "y":1},
                {"matrix": [1, 11], "x":11.5, "y":1},
                {"matrix": [1, 12], "x":12.5, "y":1},
                {"matrix": [1, 13], "x":13.5, "y":1, "w":1.5},
                {"matrix": [1, 14], "x":15, "y":1},

                {"matrix": [2, 0], "x":0, "y":2, "w":1.75},
                {"matrix": [2, 1], "x":1.75, "y":2},
                {"matrix": [2, 2], "x":2.75, "y":2},
                {"matrix": [2, 3], "x":3.75, "y":2},
                {"matrix": [2, 4], "x":4.75, "y":2},
                {"matrix": [2, 5], "x":5.75, "y":2},
                {"matrix": [2, 6], "x":6.75, "y":2},
                {"matrix": [2, 7], "x":7.75, "y":2},
                {"matrix": [2, 8], "x":8.75, "y":2},
                {"matrix": [2, 9], "x":9.75, "y":2},
                {"matrix": [2, 10], "x":10.75,"y":2},
                {"matrix": [2, 11], "x":11.75,"y":2},
                {"matrix": [2, 13], "x":12.75,"y":2, "w":2.25},
                {"matrix": [3, 14], "x":15, "y":2},

                {"matrix": [3, 0], "x":0, "y":3, "w":2.25},
                {"matrix": [3, 2], "x":2.25, "y":3},
                {"matrix": [3, 3], "x":3.25, "y":3},
                {"matrix": [3, 4], "x":4.25, "y":3},
                {"matrix": [3, 5], "x":5.25, "y":3},
                {"matrix": [3, 6], "x":6.25, "y":3},
                {"matrix": [3, 7], "x":7.25, "y":3},
                {"matrix": [3, 8], "x":8.25, "y":3},
                {"matrix": [3, 9], "x":9.25, "y":3},
                {"matrix": [3, 10], "x":10.25,"y":3},
                {"matrix": [3, 11], "x":11.25,"y":3},
                {"matrix": [3, 12], "x":12.25,"y":3, "w":1.75},
                {"matrix": [2, 14], "x":14, "y":3},
                {"matrix": [4, 14], "x":15, "y":3},

                {"matrix": [4, 0], "x":0, "y":4, "w":1.25},
                {"matrix": [4, 1], "x":1.25, "y":4, "w":1.25},
                {"matrix": [4, 2], "x":2.5, "y":4, "w":1.25},
                {"matrix": [4, 4], "x":3.75, "y":4, "w":3},
                {"matrix": [4, 5], "x":3.75, "y":4, "w":6.25},
                {"matrix": [4, 6], "x":6.75, "y":4, "w":3},
                {"matrix": [4, 9], "x":9.75, "y":4, "w":1.25},
                {"matrix": [4, 10], "x":11.25,"y":4, "w":1.25},
                {"matrix": [4, 13], "x":13, "y":4},
                {"matrix": [4, 12], "x":14, "y":4},
                {"matrix": [3, 13], "x":15, "y":4}
            ]
        },
```

**Review by dunk2k — CHANGES_REQUESTED** [review]

File: keyboards/melgeek/z70ultra_v1/keymaps/via/keymap.c [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+        _______,          _______, _______, _______, _______, _______, NK_TOGG, _______, _______, _______, _______, _______, KC_VOLU, _______,
+        _______, _______, _______,          	     _______, 				 _______, _______,          KC_MPRV, KC_VOLD, KC_MNXT
+    ),
+	[2] = LAYOUT_65_ansi_blocker( /* MO(2) */
+        _______, KC_BRID, KC_BRIU, KC_MCTL, KC_LPAD, RGB_VAD, RGB_VAI, KC_MPRV, KC_MPLY, KC_MNXT, KC_MUTE, KC_VOLD, KC_VOLU, _______, _______,
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______, _______,
+        _______,          _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
+        _______, _______, _______,		     _______,			         _______, _______,          _______, _______, _______ 
+	),
```
Comment (dunk2k) [inline keyboards/melgeek/z70ultra_v1/keymaps/via/keymap.c:39]:
```suggestion
    [0] = LAYOUT_all( /* Base */
        QK_GESC, KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC, KC_HOME,
        KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC, KC_BSLS, KC_DEL,
        KC_CAPS, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT,          KC_ENT,  KC_PGUP,
        KC_LSFT,          KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_RSFT, KC_UP,   KC_PGDN,
        KC_LCTL, KC_LGUI, KC_LALT,        KC_SPC,       KC_SPC,        KC_SPC,           MO(1),   MO(2),            KC_LEFT, KC_DOWN, KC_RGHT
    ),
    [1] = LAYOUT_all( /* MO(1) */
        KC_GRV,  KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,  KC_DEL,  _______,
        _______, RGB_TOG, RGB_MOD, RGB_HUI, RGB_HUD, RGB_SAI, RGB_SAD, RGB_VAI, RGB_VAD, _______, KC_PSCR, KC_SCRL, KC_PAUS, QK_BOOT, _______,
        _______, RGB_SPI, RGB_SPD, _______, _______, _______, _______, _______, _______, _______, _______, _______,          EE_CLR,  _______,
        _______,          _______, _______, _______, _______, _______, NK_TOGG, _______, _______, _______, _______, _______, KC_VOLU, _______,
        _______, _______, _______,        _______,      _______,       _______,          _______, _______,          KC_MPRV, KC_VOLD, KC_MNXT
    ),
    [2] = LAYOUT_all( /* MO(2) */
        _______, KC_BRID, KC_BRIU, KC_MCTL, KC_LPAD, RGB_VAD, RGB_VAI, KC_MPRV, KC_MPLY, KC_MNXT, KC_MUTE, KC_VOLD, KC_VOLU, _______, _______,
        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______, _______,
        _______,          _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
        _______, _______, _______,        _______,      _______,       _______,          _______, _______,          _______, _______, _______ 
    )
```

**Review by dunk2k — CHANGES_REQUESTED** [review]

File: keyboards/melgeek/z70ultra_v1/keymaps/default/keymap.c [hunk 10]
```diff
@@ -0,0 +1,42 @@
...
+        _______,          _______, _______, _______, _______, _______, NK_TOGG, _______, _______, _______, _______, _______, KC_VOLU, _______,
+        _______, _______, _______,          _______, _______, _______,                   _______, _______,          KC_MPRV, KC_VOLD, KC_MNXT
+    ),
+	[2] = LAYOUT_65_ansi_blocker( /* MO(2) */
+        _______, KC_BRID, KC_BRIU, KC_MCTL, KC_LPAD, RGB_VAD, RGB_VAI, KC_MPRV, KC_MPLY, KC_MNXT, KC_MUTE, KC_VOLD, KC_VOLU, _______, _______,
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______, _______,
+        _______,          _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
+        _______, _______, _______,			_______, _______, _______,					 _______, _______,          _______, _______, _______ 
+	),
```
Comment (dunk2k) [inline keyboards/melgeek/z70ultra_v1/keymaps/default/keymap.c:40]: fixes CI build error for this keymap
```suggestion
    [0] = LAYOUT_all( /* Base */
        QK_GESC, KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC, KC_HOME,
        KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC, KC_BSLS, KC_DEL,
        KC_CAPS, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT,          KC_ENT,  KC_PGUP,
        KC_LSFT,          KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_RSFT, KC_UP,   KC_PGDN,
        KC_LCTL, KC_LGUI, KC_LALT,			KC_SPC,	 KC_SPC,  KC_SPC,                  	 MO(1),   MO(2),            KC_LEFT, KC_DOWN, KC_RGHT
    ),
    [1] = LAYOUT_all( /* MO(1) */
        KC_GRV,  KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,  KC_DEL,  _______,
        _______, RGB_TOG, RGB_MOD, RGB_HUI, RGB_HUD, RGB_SAI, RGB_SAD, RGB_VAI, RGB_VAD, _______, KC_PSCR, KC_SCRL, KC_PAUS, QK_BOOT, _______,
        _______, RGB_SPI, RGB_SPD, _______, _______, _______, _______, _______, _______, _______, _______, _______,          EE_CLR,  _______,
        _______,          _______, _______, _______, _______, _______, NK_TOGG, _______, _______, _______, _______, _______, KC_VOLU, _______,
        _______, _______, _______,          _______, _______, _______,                   _______, _______,          KC_MPRV, KC_VOLD, KC_MNXT
    ),
	[2] = LAYOUT_all( /* MO(2) */
        _______, KC_BRID, KC_BRIU, KC_MCTL, KC_LPAD, RGB_VAD, RGB_VAI, KC_MPRV, KC_MPLY, KC_MNXT, KC_MUTE, KC_VOLD, KC_VOLU, _______, _______,
        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______, _______,
        _______,          _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
        _______, _______, _______,			_______, _______, _______,					 _______, _______,          _______, _______, _______ 
	)
```

File: keyboards/melgeek/z70ultra_v1/info.json [hunk 11]
```diff
@@ -0,0 +1,391 @@
...
+    "url": "",
+    "maintainer": "melgeek001365",
+    "usb": {
+        "vid": "0xEDED",
+        "pid": "0x6575",
+        "device_version": "0.0.1"
+    },
+	"processor": "STM32F303",
+    "bootloader": "stm32-dfu",
+    "board": "QMK_PROTON_C",
```
Comment (drashna) [inline keyboards/melgeek/z70ultra_v1/info.json:13]: Is this actually using a proton C? 

Given availability of them, I'm assuming no.   And if that is the case, should avoid using the board files for it, and use the default board files (and enable whichever features you need)

File: keyboards/melgeek/z70ultra_v1/info.json [hunk 12]
```diff
@@ -0,0 +1,391 @@
...
+	"diode_direction": "COL2ROW",
+	"debounce": 3,
+	"dynamic_keymap": {
+        "layer_count": 4
+    },
```
Comment (drashna) [inline keyboards/melgeek/z70ultra_v1/info.json:18]: This isn't actually needed here, as 4 is the default value.
```suggestion
```

File: keyboards/melgeek/z70ultra_v1/info.json
Comment (drashna) [inline keyboards/melgeek/z70ultra_v1/info.json:1]: You have a mix of spaces and tabs here.  Could you fix that?  (`qmk format-json` can help with that)

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/melgeek/z70ultra_v1/rev1/rev1.c [hunk 13]
```diff
@@ -0,0 +1,145 @@
...
+        RGB_MATRIX_INDICATOR_SET_COLOR(SCROLL_LOCK_INDEX_RIGHT, 0, 0, 0);
+    }
+	
+    return false;
+}
+
+void matrix_init_kb(void) {
+    for (int i = SCROLL_LOCK_INDEX_LEFT; i <= SCROLL_LOCK_INDEX_RIGHT; ++i) {
+        is31_led led = g_is31_leds[i];
+        IS31FL3741_set_scaling_registers(&led, 0xFF, 0xFF, 0xFF);
```
Comment (drashna) [inline keyboards/melgeek/z70ultra_v1/rev1/rev1.c:138]:
```suggestion
        is31fl3741_set_scaling_registers(&led, 0xFF, 0xFF, 0xFF);
```

File: keyboards/melgeek/z70ultra_v1/rev1/rev1.c [hunk 14]
```diff
@@ -0,0 +1,145 @@
...
+    }
+
+    IS31FL3741_update_led_control_registers(DRIVER_ADDR_1, 0);
```
Comment (drashna) [inline keyboards/melgeek/z70ultra_v1/rev1/rev1.c:141]:
```suggestion
    is31fl3741_update_led_control_registers(DRIVER_ADDR_1, 0);
```

**Review by drashna — CHANGES_REQUESTED** [review]
led driver names have changed, please update:

**Review by drashna — DISMISSED** [review]

File: keyboards/melgeek/z70ultra_v1/info.json [hunk 15]
```diff
@@ -0,0 +1,388 @@
...
+    "manufacturer": "MelGeek",
+    "url": "",
+    "maintainer": "melgeek001365",
+    "usb": {
+        "vid": "0xEDED",
+        "pid": "0x6575",
+        "device_version": "0.0.1"
+    },
+    "processor": "STM32F303",
+    "bootloader": "stm32-dfu",
```
Comment (dunk2k) [inline keyboards/melgeek/z70ultra_v1/info.json:12]: fixes CI Build failure
```suggestion
    "bootloader": "stm32-dfu",
    "board": "QMK_PROTON_C",
```

**Review by dunk2k — CHANGES_REQUESTED** [review]

File: keyboards/melgeek/z70ultra_v1/info.json (hunk 15 above)
Comment (drashna) [reply keyboards/melgeek/z70ultra_v1/info.json:12]: This isn't the proper way to solve the issue.   Unless the board is using a proton C, it shouldn't use the proton c board files.  Instead, halconf. and mcuconf.h should be provided to enable the needed hardware peripharals.

Comment (mj-bomb) [reply keyboards/melgeek/z70ultra_v1/info.json:12]: Thank you for your suggestion, but how can I modify it? Are there any examples for reference?

Comment (drashna) [reply keyboards/melgeek/z70ultra_v1/info.json:12]: Yeah, np.  Mostly, it's figure out what you need, and enable that.  The moonlander was initially added using the board files, and you can see it migrated completely off of it, in this commit: 
https://github.com/qmk/qmk_firmware/commit/40578621a76b43481116fd73b10b8cdec3d3a7ef

basically, you need the halconf.h and mcuconf.h files to configure things.  And you should only enable what you need to.

File: keyboards/melgeek/z70ultra_v1/info.json [hunk 16]
```diff
@@ -0,0 +1,389 @@
...
+                {"matrix": [2, 14], "x":14,    "y":3},
+                {"matrix": [4, 14], "x":15,    "y":3},
+
+                {"matrix": [4, 0],  "x":0,     "y":4, "w":1.25},
+                {"matrix": [4, 1],  "x":1.25,  "y":4, "w":1.25},
+                {"matrix": [4, 2],  "x":2.5,   "y":4, "w":1.25},
+                {"matrix": [4, 4],  "x":3.75,  "y":4, "w":3},
+                {"matrix": [4, 5],  "x":3.75,  "y":4, "w":6.25},
+                {"matrix": [4, 6],  "x":6.75,  "y":4, "w":3},
+                {"matrix": [4, 9],  "x":9.75,  "y":4, "w":1.25},
```
Comment (noroadsleft) [inline keyboards/melgeek/z70ultra_v1/info.json:229]: Fixes some key overlap issues in QMK Configurator.

image

```suggestion
                {"matrix": [4, 4],  "x":3.75,  "y":4.5, "w":3, "h": 0.5},
                {"matrix": [4, 5],  "x":3.75,  "y":4, "w":6, "h": 0.5},
                {"matrix": [4, 6],  "x":6.75,  "y":4.5, "w":3, "h": 0.5},
                {"matrix": [4, 9],  "x":9.75,  "y":4, "w":1.5},
```

**Review by noroadsleft — CHANGES_REQUESTED** [review]
Additionally, I'm not a fan of the `z70ultra_v1/rev1` name - to my mind, `v1` means the same thing as `rev1`.

I'd suggest using `melgeek/z70ultra/v1` or `melgeek/z70ultra/rev1` instead

File: keyboards/melgeek/z70ultra_v1/config.h [hunk 17]
```diff
@@ -0,0 +1,33 @@
...
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#define RGB_DISABLE_WHEN_USB_SUSPENDED // turn off effects when suspended
+#define RGB_MATRIX_KEYPRESSES
```
Comment (drashna) [inline keyboards/melgeek/z70ultra_v1/config.h:20]: Inferred by animations enabled. Not needed.
```suggestion
```

File: keyboards/melgeek/z70ultra_v1/config.h [hunk 18]
```diff
@@ -0,0 +1,33 @@
...
+#define RGB_MATRIX_LED_PROCESS_LIMIT 4
+#define RGB_MATRIX_LED_FLUSH_LIMIT 26
+
+#define RGB_MATRIX_DEFAULT_MODE RGB_MATRIX_CYCLE_ALL
+#define DRIVER_ADDR_1 0b0110000
+#define DRIVER_COUNT 1
```
Comment (drashna) [inline keyboards/melgeek/z70ultra_v1/config.h:26]:
```suggestion
#define IS31FL3741_I2C_ADDRESS_1 IS31FL3741_I2C_ADDRESS_GND
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Pull request closed without merging**